plugins/gnupgplugin/gpgme_p.h

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (C) 2018 Damien Caliste.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef _FILE_OFFSET_BITS
+#define _FILE_OFFSET_BITS 64
+#endif
+#include <gpgme.h>
+
+#include <Crypto/key.h>
+#include <QString>
+#include <QDebug>
+#include <QDateTime>
+
+struct GPGmeContext {
+    gpgme_ctx_t ctx;
+    gpgme_error_t err;
+    GPGmeContext(gpgme_protocol_t protocol, const QString &home = QString())
+        : ctx(0), err(0)
+    {
+        err = gpgme_engine_check_version(protocol);
+        if (gpgme_err_code(err) != GPG_ERR_NO_ERROR) {
+            gpgme_engine_info_t info;
+            gpgme_error_t err2;
+            err2 = gpgme_get_engine_info(&info);
+            if (!err2) {
+                while (info) {
+                    if (info->protocol == protocol) {
+                        qWarning() << "protocol:" << gpgme_get_protocol_name(info->protocol);
+                        if (info->file_name && !info->version)
+                            qWarning() << "engine " << info->file_name
+                                       << " not installed properly";
```

Comment:
I believe `QDebug` is inserting spaces between the strings, there's no need to add more.

Reply:
Yep.